airflow/models/taskinstance.py

```diff
@@ -3673,21 +3673,12 @@ def _schedule_downstream_tasks(
                 assert task
                 assert task.dag
 
-            # Get a partial DAG with just the specific tasks we want to examine.
-            # In order for dep checks to work correctly, we include ourself (so
-            # TriggerRuleDep can check the state of the task we just executed).
-            partial_dag = task.dag.partial_subset(
-                task.downstream_task_ids,
-                include_downstream=True,
-                include_upstream=False,
-                include_direct_upstream=True,
-            )
-
-            dag_run.dag = partial_dag
+            # In many scenarios, get a partial DAG is an unsafe operation and will lead to many bad cases.
```

Comment:
Last nit - someone who sees this comment for the first will need some context to understand what's going on. If you think that this might require additional attention in the future, please state it explicitly and refer to this PR. Otherwise, this comment will have no use and should be removed.

Reply:
Thank you for the suggestion. I've updated the comment to provide better context.